aptos-move/aptos-aggregator/src/aggregator_extension.rs

```diff
@@ -492,8 +486,117 @@ impl AggregatorData {
         Ok(snapshot_id)
     }
 
-    pub fn read_snapshot(&self, _id: AggregatorVersionedID) -> PartialVMResult<u128> {
-        unimplemented!();
+    pub fn create_new_snapshot(&mut self, id: AggregatorID, value: SnapshotValue) {
+        let snapshot_state = AggregatorSnapshotState::Create { value };
+
+        self.aggregator_snapshots.insert(id, AggregatorSnapshot {
+            id,
+            state: snapshot_state,
+        });
+    }
+
+    /// Returns a reference to an AggregatorSnapshot with `id`.
+    /// If transaction that is currently executing did not initialize it,
+    /// a new AggregatorSnapshot instance is created.
+    /// Note: when we say "aggregator snapshot instance" here we refer to Rust struct and
+    /// not to the Move aggregator snapshot.
+    pub fn get_snapshot(
+        &mut self,
+        id: AggregatorID,
+        resolver: &dyn AggregatorResolver,
+    ) -> PartialVMResult<&AggregatorSnapshot> {
+        let snapshot = match self.aggregator_snapshots.entry(id) {
+            Entry::Vacant(entry) => {
+                // Otherwise, we have to go to storage and read the value.
+                let value_from_storage = resolver
+                    .get_aggregator_v2_value(&id, AggregatorReadMode::Aggregated)
```

Comment:
`get_aggregator_v2_value` returns only u128. What if the snapshot is a String snapshot?

Reply:
dependent aggregator can only be u128. only derived snapshot generats string output, string is never input